iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/CalculatedBusImpl.java

```diff
@@ -26,9 +24,38 @@
 
     private final List<NodeTerminal> terminals;
 
-    CalculatedBusImpl(String id, VoltageLevelExt voltageLevel, List<NodeTerminal> terminals) {
+    private NodeTerminal terminalRef;
+
+    CalculatedBusImpl(String id, NodeBreakerVoltageLevel voltageLevel, TIntArrayList nodes, List<NodeTerminal> terminals) {
         super(id, voltageLevel);
         this.terminals = Objects.requireNonNull(terminals);
+        this.terminalRef = findTerminal(voltageLevel, nodes, terminals);
+    }
+
+    private static NodeTerminal findTerminal(NodeBreakerVoltageLevel voltageLevel, TIntArrayList nodes, List<NodeTerminal> terminals) {
+        if (!terminals.isEmpty()) {
+            return terminals.get(0);
+        }
+
+        NodeTerminal[] terminal = new NodeTerminal[1];
+
+        // Traverse the graph until a valid NodeTerminal is found
+        VoltageLevel.NodeBreakerView.Traverser traverser = (node1, sw, node2) -> {
+            if (sw.isOpen()) {
+                return false;
+            }
+            terminal[0] = (NodeTerminal) voltageLevel.getNodeBreakerView().getTerminal(node2);
+            return terminal[0] == null;
+        };
+
+        for (int i = 0; i < nodes.size(); ++i) {
```

Comment:
Not very important, but I think only the first traversal is useful ?
If a terminal can be found, it will be found on first traversal or not found at all.

Reply:
Agree: I think the for-loop is useless, but I had a doubt when I wrote the code. I will remove the loop: if there is no bug in the traverser, we should find a terminal.